### DISCUSSION TITLE
Pop/Shift Multiple values [Question]

### DISCUSSION THREAD

**Discussion by thoagland** [rlogiacco/CircularBuffer#33] [Q&A] [answered]
Hello,

I've been experimenting wit this library and love the simplicity of the interface.  I was wondering it is is possible to "retrieve" large sections of the buffer at a time.  For example, if my data collection loop has been recording data, once a second, grab all[most] of the current data in one call with something like:
>buffer.pop(buffer.size())

Is this type of functionality available?  [Sorry if I missed where this has been previously discussed.]

-Tim

**Comment by rlogiacco** [comment, marked as answer]
I fail to understand the reason for willing to do so, anyway, the answer is no, there is no operation to extract all elements of this data structure into some other data structure.
